### ISSUE TITLE
#8271 Verify Local Data Wipes Out [Deletes] Entire Data File *previously completed

### ISSUE THREAD

**Issue by chaseholden** [transmission/transmission#3716]
#8271 build 9e07b6f4c5 
macOS 12.5.1 Monterey on M1 Ultra Mac Studio 64GB 4TB

Verify Local Data Wipes Out [Deletes] Entire Data File and progress* if download was completed in previous build version!!!

Apparently, those files which are completed in this current version remain intact after Verify Local Data operation, but I lost quit a bit of data and files because of this bug. 

Note that I see limited success running Verify Local Data operation a second consecutive time in which more data is revealed completed according to progress bar and percentage and some files are then shown as complete again second time.

**Comment by ckerr** [comment]
@chaseholden walk me through how to repro this with a clean build of 9e07b6f and an earlier build of your choice

1. Tell me what earlier build to use as the control
2. Let's say I download an Ubuntu iso with the earlier build, then upgrade to the newer build. Are you saying the .iso file will be wiped out, or ...? What specific steps should I follow to trigger this bug?

**Comment by ckerr** [comment]
ping?

**Comment by chaseholden** [comment]
@ckerr I apologize for the delayed response while AFK…

What I was stating happened is that if you had downloaded an Ubuntu.ISO file that was LESS than 100% complete*, like say 96% complete*, then quit, launched updated version of Transmission, then that Ubuntu.ISO file would start back at 0% or at least display 0% in Transmission's progress percentage even after performing the Validate Local Data Operation.

* Note that I do utilize Transmission's ability to have a separate Directory for incomplete downloads and another Directory for Completed files. Transmission always successfully moves files upon completion from the incomplete Dir I call "/bits/" to the parent directory upon completion.

I saved the Application Support dir for Transmission for that version but I don't know what that previous build number was other than that it was dated 8/23/2022 and I used it until 8/26/2022 at which point I updated to this #8271 build 9e07b6f 
It doesn't look like anything in that directory contains a record of the build or version. 

If it helps, I can provide you with the backup /App Support dir for that previous version and the version after it where the wipeout occurred.
Does this help?
